jvb/src/main/kotlin/org/jitsi/videobridge/JettyHelpers.kt

```diff
@@ -0,0 +1,130 @@
+@file:JvmName("JettyHelpers")
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.videobridge
+
+import org.eclipse.jetty.server.HttpConfiguration
+import org.eclipse.jetty.server.HttpConnectionFactory
+import org.eclipse.jetty.server.SecureRequestCustomizer
+import org.eclipse.jetty.server.Server
+import org.eclipse.jetty.server.ServerConnector
+import org.eclipse.jetty.server.SslConnectionFactory
+import org.eclipse.jetty.servlet.ServletContextHandler
+import org.eclipse.jetty.servlet.ServletHolder
+import org.eclipse.jetty.util.ssl.SslContextFactory
+import org.jitsi.rest.JettyBundleActivatorConfig
+import java.nio.file.Paths
+
+/**
+ * Create a non-secure Jetty server instance listening on the given [port] and [host] address.
+ */
+fun createJettyServer(
+    port: Int,
+    host: String? = null
+): Server {
+    val server = Server().apply {
+        handler = ServletContextHandler()
+    }
+    val connector = ServerConnector(server, HttpConnectionFactory(HttpConfiguration())).apply {
+        this.port = port
+        this.host = host
+    }
+    server.addConnector(connector)
+    return server
+}
+
+/**
+ * Create a secure Jetty server instance listening on the given [port] and [host] address and using the
+ * KeyStore located at [keyStorePath], optionally protected by [keyStorePassword].  [needClientAuth] sets whether
+ * client auth is needed for SSL (see [SslContextFactory.setNeedClientAuth]).
+ */
+fun createSecureJettyServer(
+    port: Int,
+    keyStorePath: String,
+    host: String? = null,
+    keyStorePassword: String? = null,
+    needClientAuth: Boolean = false
+): Server {
+    val sslContextFactoryKeyStoreFile = Paths.get(keyStorePath).toFile()
+    val sslContextFactory = SslContextFactory().apply {
+        setIncludeProtocols("TLSv1.2")
```

Comment:
jicoco adds "TLSv1.3" on java >= 11. Do we want the same here?

Reply:
I left that out because it seemed unnecessary and determining the Java version was a little awkward, from what I remember.

Reply:
cc @JonathanLennox, he specifically added it recently

Reply:
Thanks for pointing that out, I didn't realized it had been added recently.  It doesn't look like it's a Java 11 compatibility thing so much as opportunistically doing TLS 1.3.  My hope was that we could ignore this for now and, when it comes up, maybe be in a better place about java 11 support and therefore avoid the hacky java version check, but I dunno.  Will let Jonathan weigh in.

Reply:
TLS 1.3 is only supported on Java 11+, which is why the condition, and I believe the `setIncludeProtocols` call fails on older Java for unknown versions.

Getting the version is only hacky because of the syntax change ("1.8" -> "11") which technically you could ignore anyway since this only cares about 11+.

Reply:
I think the question is: do we need to be opportunistically enabling TLS 1.3 for Java 11+ (which we don't currently use anywhere)?  What prompted adding that?

Reply:
Just completeness -- I wanted to match the Mozilla Guidelines config recommendations as closely as possible.